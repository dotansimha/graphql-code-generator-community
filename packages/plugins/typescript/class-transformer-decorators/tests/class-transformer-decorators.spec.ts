import { buildSchema } from 'graphql';
import { mergeOutputs } from '@graphql-codegen/plugin-helpers';
import { plugin } from '../src/index.js';

const SCHEMA = `
  type User {
    id: ID!
    name: String!
  }

  type Address {
    id: String
    city: String!
    postalCode: String!
    stateProvince: String!
    street: String!
  }

  input AddressInput {
    id: String
    city: String!
    postalCode: String!
    stateProvince: String!
    street: String!
  }

  input UserInput {
    id: String
    name: String
    address: AddressInput
  }
`;

describe('class-transformer-decorators', () => {
  it('should decorate @Expose() on matching class properties and @Type on all nested classes', async () => {
    const schema = buildSchema(SCHEMA);
    const result = mergeOutputs([
      await plugin(schema, [], {
        classWhitelist: ['User'],
        classNamePattern: '.*Input$',
        declarationKind: 'class',
      }),
    ]);

    expect(result).toMatchSnapshot();
  });
});
