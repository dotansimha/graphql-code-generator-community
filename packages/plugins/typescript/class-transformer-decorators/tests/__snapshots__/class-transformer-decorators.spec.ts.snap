// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class-transformer-decorators should decorate @Expose() on matching class properties and @Type on all nested classes 1`] = `
"import 'reflect-metadata';
import { Expose, Type } from 'class-transformer';
export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
export type MakeEmpty<T extends { [key: string]: unknown }, K extends keyof T> = { [_ in K]?: never };
export type Incremental<T> = T | { [P in keyof T]?: P extends ' $fragmentName' | '__typename' ? T[P] : never };
/** All built-in and custom scalars, mapped to their actual values */
export class Scalars {
  ID: {
    input: string;
    output: string;
  };
  String: {
    input: string;
    output: string;
  };
  Boolean: {
    input: boolean;
    output: boolean;
  };
  Int: {
    input: number;
    output: number;
  };
  Float: {
    input: number;
    output: number;
  };
}
;
export class User {
  @Expose()
  __typename?: 'User';
  @Expose()
  id: Scalars['ID']['output'];
  @Expose()
  name: Scalars['String']['output'];
}
;
export class Address {
  __typename?: 'Address';
  id?: Maybe<Scalars['String']['output']>;
  city: Scalars['String']['output'];
  postalCode: Scalars['String']['output'];
  stateProvince: Scalars['String']['output'];
  street: Scalars['String']['output'];
}
;
export class AddressInput {
  @Expose()
  id?: InputMaybe<Scalars['String']['input']>;
  @Expose()
  city: Scalars['String']['input'];
  @Expose()
  postalCode: Scalars['String']['input'];
  @Expose()
  stateProvince: Scalars['String']['input'];
  @Expose()
  street: Scalars['String']['input'];
}
;
export class UserInput {
  @Expose()
  id?: InputMaybe<Scalars['String']['input']>;
  @Expose()
  name?: InputMaybe<Scalars['String']['input']>;
  @Expose()
  @Type(() => AddressInput)
  address?: InputMaybe<AddressInput>;
}
;"
`;
